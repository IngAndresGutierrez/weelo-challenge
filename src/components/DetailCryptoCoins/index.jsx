/* import external modules */
import { List, ListItem, ListItemText } from '@material-ui/core'

/* import internal modules */
import { useStyles } from './styles'
import {
  addCurrencyFormatUsd,
  addCurrencyFormatUsdWithoutDollarSymbol,
} from '../../utils/currencyFormat'

/** @description This is a functional component for detail about cryptocurreny coins
 *  @version 1.0.0
 *  @since 11/14/2021
 *  @author Jaime Andr칠s G칩mez Guti칠rrez <g.gutierrez.j.andres@gmail.com>
 *
 *  @function
 *  @name DetailCryptoCoins
 *  @returns {Component} Returns the detail about cryptocurreny coins component
 **/

const DetailCryptoCoins = ({ informationCoin }) => {
  const classes = useStyles()

  const renderDetailCryptoCoin = (
    <>
      <ListItem divider>
        <ListItemText secondary={`Symbol 游목`} />
        <ListItemText primary={`${informationCoin?.symbol}`} />
      </ListItem>
      <ListItem divider>
        <ListItemText secondary={`Name 游목`} />
        <ListItemText primary={`${informationCoin?.name}`} />
      </ListItem>
      <ListItem divider>
        <ListItemText secondary={`Rank 游목`} />
        <ListItemText primary={`${informationCoin?.rank}`} />
      </ListItem>
      <ListItem divider>
        <ListItemText secondary={`Price USD 游목  `} />
        <ListItemText
          primary={`${addCurrencyFormatUsd(informationCoin?.price_usd)}`}
        />
      </ListItem>
      <ListItem divider>
        <ListItemText secondary={`% Change 24h 游목`} />
        <ListItemText primary={`${informationCoin?.percent_change_24h}`} />
      </ListItem>
      <ListItem divider>
        <ListItemText secondary={`% Change 1h 游목`} />
        <ListItemText primary={`${informationCoin?.percent_change_1h}`} />
      </ListItem>
      <ListItem divider>
        <ListItemText secondary={`% Change 7D 游목`} />
        <ListItemText primary={`${informationCoin?.percent_change_7d}`} />
      </ListItem>
      <ListItem divider>
        <ListItemText secondary={`Price BTC 游목`} />
        <ListItemText primary={`${informationCoin?.price_btc}`} />
      </ListItem>
      <ListItem divider>
        <ListItemText secondary={`Market Cap USD 游목`} />
        <ListItemText
          primary={`${addCurrencyFormatUsd(informationCoin?.market_cap_usd)}`}
        />
      </ListItem>
      <ListItem divider>
        <ListItemText secondary={`Volume24 游목`} />
        <ListItemText
          primary={`${addCurrencyFormatUsdWithoutDollarSymbol(
            informationCoin?.volume24
          )}`}
        />
      </ListItem>
      <ListItem divider>
        <ListItemText secondary={`Volume24A 游목`} />
        <ListItemText
          primary={`${addCurrencyFormatUsdWithoutDollarSymbol(
            informationCoin?.volume24a
          )}`}
        />
      </ListItem>
      <ListItem divider>
        <ListItemText secondary={`Csupply 游목`} />
        <ListItemText
          primary={`${addCurrencyFormatUsdWithoutDollarSymbol(
            informationCoin?.csupply
          )}`}
        />
      </ListItem>
      <ListItem divider>
        <ListItemText secondary={`Tsupply 游목`} />
        <ListItemText
          primary={`${addCurrencyFormatUsdWithoutDollarSymbol(
            informationCoin?.tsupply
          )}`}
        />
      </ListItem>
      <ListItem divider>
        <ListItemText secondary={`Msupply 游목`} />
        <ListItemText
          primary={`${addCurrencyFormatUsdWithoutDollarSymbol(
            informationCoin?.msupply
          )}`}
        />
      </ListItem>
    </>
  )

  return (
    <List component="nav" className={classes.root} aria-label="detail coins">
      {renderDetailCryptoCoin}
    </List>
  )
}

export default DetailCryptoCoins
