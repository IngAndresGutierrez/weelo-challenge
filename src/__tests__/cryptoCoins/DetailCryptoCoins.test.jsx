/* import external modules */
import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'

/* import internal modules */
import '../../../setupTests'
import DetailCryptoCoins from '../../components/DetailCryptoCoins'

const informationCoin = {
  csupply: '18851540.00',
  id: '90',
  market_cap_usd: '1220750619767.60',
  msupply: '21000000',
  name: 'Bitcoin',
  nameid: 'bitcoin',
  percent_change_1h: '0.23',
  percent_change_7d: '6.51',
  percent_change_24h: '1.65',
  price_btc: '1.00',
  price_usd: '64756.02',
  rank: 1,
  symbol: 'BTC',
  tsupply: '18851540',
  volume24: 22726854213.71473,
  volume24a: 27873040814.98264,
}

describe('Testing suite DetailCryptoCoins component', () => {
  const DetailCryptoCoinsShallow = shallow(<DetailCryptoCoins />)
  const DetailCryptoCoinsMount = mount(
    <DetailCryptoCoins informationCoin={informationCoin} />
  )

  test('Render DetailCryptoCoins', () => {
    expect(DetailCryptoCoinsShallow).toHaveLength(1)
  })

  test('Renders correctly - Snapshot', () => {
    const treeDetailCryptoCoinsMount = renderer
      .create(DetailCryptoCoinsMount)
      .toJSON()
    expect(treeDetailCryptoCoinsMount).toMatchSnapshot()
  })

  test('Render information', () => {
    expect(DetailCryptoCoinsMount.find('span').first().text()).toEqual('BTC')
    expect(DetailCryptoCoinsMount.find('span').at(2).text()).toEqual('1')
    expect(DetailCryptoCoinsMount.find('span').at(3).text()).toEqual(
      '$64,756.02'
    )
    expect(DetailCryptoCoinsMount.find('span').at(4).text()).toEqual('1.65')
    expect(DetailCryptoCoinsMount.find('span').at(5).text()).toEqual('0.23')
  })
})
