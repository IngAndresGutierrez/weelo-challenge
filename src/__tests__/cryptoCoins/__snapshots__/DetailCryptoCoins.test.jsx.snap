// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing suite DetailCryptoCoins component Renders correctly - Snapshot 1`] = `
<nav
  aria-label="detail coins"
  className="MuiList-root makeStyles-root-1 MuiList-padding"
>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Symbol 游목
      </p>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        BTC
      </span>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Name 游목
      </p>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        Bitcoin
      </span>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Rank 游목
      </p>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        1
      </span>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Price USD 游목  
      </p>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        $64,756.02
      </span>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        % Change 24h 游목
      </p>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        1.65
      </span>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        % Change 1h 游목
      </p>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        0.23
      </span>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        % Change 7D 游목
      </p>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        6.51
      </span>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Price BTC 游목
      </p>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        1.00
      </span>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Market Cap USD 游목
      </p>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        $1,220,750,619,767.60
      </span>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Volume24 游목
      </p>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        22,726,854,213.715
      </span>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Volume24A 游목
      </p>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        27,873,040,814.983
      </span>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Csupply 游목
      </p>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        18,851,540
      </span>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Tsupply 游목
      </p>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        18,851,540
      </span>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Msupply 游목
      </p>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        21,000,000
      </span>
    </div>
  </li>
</nav>
`;
